indexing-service/src/test/java/org/apache/druid/indexing/common/task/ClientCompactionTaskQuerySerdeTest.java

```diff
@@ -71,46 +73,115 @@
 
 import java.io.IOException;
 import java.util.HashMap;
+import java.util.Map;
 
 public class ClientCompactionTaskQuerySerdeTest
 {
+  static {
+    NullHandling.initializeForTests();
+  }
+
   private static final RowIngestionMetersFactory ROW_INGESTION_METERS_FACTORY =
       new TestUtils().getRowIngestionMetersFactory();
   private static final CoordinatorClient COORDINATOR_CLIENT = new NoopCoordinatorClient();
   private static final AppenderatorsManager APPENDERATORS_MANAGER = new TestAppenderatorsManager();
+  private static final ObjectMapper MAPPER = setupInjectablesInObjectMapper(new DefaultObjectMapper());
+
+  private static final IndexSpec INDEX_SPEC = IndexSpec.builder()
+                                                       .withDimensionCompression(CompressionStrategy.LZ4)
+                                                       .withMetricCompression(CompressionStrategy.LZF)
+                                                       .withLongEncoding(LongEncodingStrategy.LONGS)
+                                                       .build();
+  private static final IndexSpec INDEX_SPEC_FOR_INTERMEDIATE_PERSISTS = IndexSpec.builder()
+                                                                                 .withDimensionCompression(CompressionStrategy.LZ4)
+                                                                                 .withMetricCompression(CompressionStrategy.UNCOMPRESSED)
+                                                                                 .withLongEncoding(LongEncodingStrategy.AUTO)
+                                                                                 .build();
+  private static final ClientCompactionIOConfig CLIENT_COMPACTION_IO_CONFIG = new ClientCompactionIOConfig(
+      new ClientCompactionIntervalSpec(
+          Intervals.of("2019/2020"),
+          "testSha256OfSortedSegmentIds"),
+      true
+  );
+  private static final ClientCompactionTaskGranularitySpec CLIENT_COMPACTION_TASK_GRANULARITY_SPEC =
+      new ClientCompactionTaskGranularitySpec(Granularities.DAY, Granularities.HOUR, true);
+  private static final ClientCompactionTaskDimensionsSpec CLIENT_COMPACTION_TASK_DIMENSIONS_SPEC =
+      new ClientCompactionTaskDimensionsSpec(DimensionsSpec.getDefaultSchemas(ImmutableList.of("ts", "dim")));
+  private static final AggregatorFactory[] METRICS_SPEC = new AggregatorFactory[] {new CountAggregatorFactory("cnt")};
+  private static final ClientCompactionTaskTransformSpec CLIENT_COMPACTION_TASK_TRANSFORM_SPEC =
+      new ClientCompactionTaskTransformSpec(new SelectorDimFilter("dim1", "foo", null));
+  private static final DynamicPartitionsSpec DYNAMIC_PARTITIONS_SPEC = new DynamicPartitionsSpec(100, 30000L);
+  private static final SegmentsSplitHintSpec SEGMENTS_SPLIT_HINT_SPEC = new SegmentsSplitHintSpec(new HumanReadableBytes(100000L), 10);
```

Comment:
Rather than declaring all of these constants and reusing the same values in all the tests, just add private methods `createCompactionTuningConfig` and `createCompactionTaskQuery`.

Reply:
Have created 2 private methods:
1. `createCompactionTaskQuery(String id, ClientCompactionTaskTransformSpec transformSpec)`
2. `createCompactionTask(ClientCompactionTaskTransformSpec transformSpec)`

There are still a bunch of constants left though since they occur in both of the above methods, since the tuning config is different in both of them (ClientCompactionTaskQueryTuningConfig vs TuningConfig). Have removed the constants which didn't occur more than once in the refactored code.

Hope this works.